packages/mpc-tls/prover/src/handlers/proxy.rs

```diff
@@ -0,0 +1,108 @@
+use crate::{
+    core::{
+        notarize::{notarize_request, NotarizeRequestParams},
+        utils::compute_sha256_hash,
+    },
+    proxy::{EmptyProverHandlersImpl, Header, ProxyRequest, ServerConfig, BLACKLISTED_HEADERS},
+    socket::prover::{ProverServer, TlsProof},
+};
+use actix_web::{
+    route,
+    web::{self},
+    HttpRequest, HttpResponseBuilder, Responder,
+};
+use hyper::body;
+use std::sync::Arc;
+use tracing::debug;
+use url::Url;
+
+#[route(
+    "/proxy",
+    method = "GET",
+    method = "POST",
+    method = "PUT",
+    method = "PATCH"
+)]
+// recieve the parameter for the publisher as well
+pub async fn handle_notarization_request(
+    payload: web::Payload,
+    req: HttpRequest,
+    data: web::Data<ServerConfig>,
+) -> impl Responder {
+    let config = data.get_ref();
+    let reply_handlers = Arc::new(EmptyProverHandlersImpl {});
+    let mut prover_socket = ProverServer::new(
```

Comment:
maybe it would be nice to make this `ProverServer` a parameter of `handle_notarization_request`; otherwise, it'll keep binding to the socket on each new request. Makes sense? In reality I don't know what are the penalties for binding to the sockets at each new request, just a suggestion

Reply:
agreed, but passing a complicated struct as this through acticx_web's web::Data struct is proving impossible. i will create a ticket to optimize this seperately.